### PR TITLE
Fix onnx.Pad constant

### PR DESCRIPTION
**Pull request by josel-amd**
Looking at the `GeneratedTorchOps.td`. This leaves two options, either the operator specification is wrong or the code generating it.

I check a bit and it seems that indeed the op spec is right. References:

https://github.com/pytorch/pytorch/blob/543a870943120484db547382ed9ca9538a40f284/torch/csrc/api/include/torch/nn/functional/padding.h#L12

https://pytorch.org/docs/main/generated/torch.nn.functional.pad.html#torch.nn.functional.pad

```
def Torch_AtenPadOp : Torch_Op<"aten.pad", [
    AllowsTypeRefinement,
    HasValueSemantics,
    ReadOnly
  ]> {
  let summary = "Generated op for `aten::pad : (Tensor, int[], str, float?) -> (Tensor)`";
  let arguments = (ins
    AnyTorchTensorType:$self,
    AnyTorchListOfTorchIntType:$pad,
    Torch_StringType:$mode,
    AnyTorchOptionalFloatType:$value
  );
  let results = (outs
    AnyTorchTensorType:$result
  );
  let hasCustomAssemblyFormat = 1;
  let extraClassDefinition = [{
    ParseResult AtenPadOp::parse(OpAsmParser &parser, OperationState &result) {
      return parseDefaultTorchOp(parser, result, 4, 1);
    }
    void AtenPadOp::print(OpAsmPrinter &printer) {
      printDefaultTorchOp(printer, *this, 4, 1);
    }
  }];
}
```

### CODE REVIEW

**Comment by cferry-AMD** [comment]
To be sure I understood correctly, is the support of other data types than `f32` what is wrong in `TorchOnnxToTorch`? Is there any test case where this appears?

**Comment by josel-amd** [comment]
> To be sure I understood correctly, is the support of other data types than `f32` what is wrong in `TorchOnnxToTorch`? Is there any test case where this appears?

Yes, only `f32` is allowed in that specific argument. I don't think there is a test case for it. At least nothing started failing. I can add one, if there is general agreement about the fix.

**Review by cferry-AMD — APPROVED** [review]
Good, indeed if we look closely at what happens above, any constant scalar passed is turned into a float. Thanks!

**Review by mgehre-amd — APPROVED** [review]
